ts/client/index.ts

```diff
@@ -15,6 +16,16 @@ async function main() {
     const fast = fastify({ logger: true });
     fast.setErrorHandler(console.error);
 
+    fast.addHook("onRequest", async (request, reply) => {
+        if (node.getSyncMode() === SyncMode.INITIAL_SYNCING) {
+            return reply.status(503).send({
+                message: "Initial sync incomplete",
+                error: "RPC unavailable",
+                statusCode: 503
+            });
+        }
+    });
```

Comment:
I think we can move this into `node.ts` and remove the getter for syncMode, more appropriate for node to manage this hook.

I addition, one downside of this is that all endpoints will return this status including our process health check (`/health`) which would be used by container orchestration platforms and uptime checks in general to make sure the node is still responding to pings. We should have the hook ignore `/health` or consider adding the hook only to the endpoints registered in `RPC.ts`.

Reply:
the fastify Instance is only present inside the "static" init method of the node class, adding the onRequest hook inside that function won't work at runtime as we need to check the syncMode at every API call. to make node manage the hook, fastify instance must be saved as a class variable and then the hook must be added inside the `start` method.

Reply:
You can move to node without saving instance since the `syncer` service is in scope:
![Screenshot from 2021-07-22 17-41-53](https://user-images.githubusercontent.com/10394161/126722153-90966c5c-ea65-4ee8-b5f7-d81daa50def7.png)

Reply:
oh so sorry, my bad
i was thinking that the `syncer` instance gets destroyed outside the scope of this init function. thanks 😃